### PR TITLE
Update docs to discuss exit codes for failed annotated methods

### LINKED ISSUE #497: Documentation needs to discuss exception handling for @BeforeTask, AfterTask and FailedTask
We need to document Spring Cloud Task's behavior when a @BeforeTask, @AfterTask, or @FailedTask throw an exception.

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #497

### CODE REVIEW

File: spring-cloud-task-docs/src/main/asciidoc/features.adoc [hunk 1]
```diff
@@ -314,6 +314,13 @@ https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-featu
 If so, it returns the exit code from  the `ExitCodeGenerator`. Otherwise, `1`
 is returned.
 
+In the case that an exception is thrown in an `onTaskStartup` method, the exit code for the application will be `1`.
+If an exception is thrown in either a `onTaskEnd` or `onTaskFailed`
+method, the exit code for the application will be the one established using the rules enumerated above.
+
+NOTE: In the case of an exception being thrown in a `onTaskStartup`, `onTaskEnd`, or `onTaskFailed`
+you can not override the exit code for the application using `ExitCodeExceptionMapper`.
+
```
Comment (mminella) [inline spring-cloud-task-docs/src/main/asciidoc/features.adoc]: The `onTaskStartup`, `onTaskEnd`, etc are private methods and should be considered implementation details.  Therefore we should not be referring to them directly in our docs.

Comment (cppwfs) [reply spring-cloud-task-docs/src/main/asciidoc/features.adoc]: Hmm, I'm not sure I understand, `onTaskStartup`  & `onTaskEnd` are not private as discussed here: https://docs.spring.io/spring-cloud-task/docs/current/reference/htmlsingle/#features-task-execution-listener

Comment (mminella) [reply spring-cloud-task-docs/src/main/asciidoc/features.adoc]: My mistake.  I was thinking of the do* methods on the `TaskLifecycleListener`.

**Comment by mminella** [comment]
LGTM.  Rebased and merged as b86fbee7792b648929c05c8ea4c1b87e05210d85

**Pull request closed without merging**